### PR TITLE
Migrate group icons to new icon service

### PR DESCRIPTION
**Pull request by hypeJunction**
**feature(file): adds `ElggFile::transfer()` for reliable renaming of files**
rename() is not a reliable option to renaming files on filestore, as the target file may be in a non-existent directory.

This adds a reliable and tested method to transfer an `ElggFile` to a new owner and copy file contents into a file with a new filename

**feature(groups): group icons are now handled by the new icon service**

Migrates group icon handling to the new API.
Moves icon ownership transfer logic out of the action file into an `update:after` event handler.
Moves icon deletion logic out of the delete action into the `delete` event handler.
- [x] Requires #9655

### CODE REVIEW

**Comment by hypeJunction** [comment]
I am not sure if transferring icons to a new owner has ever worked if the new owner hadn't had the `/groups` directory on filestore.

File: engine/lib/filestore.php [hunk 1]
```diff
@@ -487,6 +487,10 @@ function _elgg_filestore_init() {
 
 	// Unit testing
 	elgg_register_plugin_hook_handler('unit_test', 'system', '_elgg_filestore_test');
+
+	// Move entity icons if entity owner has changed
+	elgg_register_event_handler('update:after', 'object', '_elgg_filestore_move_icons');
```
Comment (hypeJunction) [inline engine/lib/filestore.php]: After some contemplation I decided there is room for this in core. I am sure plugins forget to do this all the time. File plugin doesn't do this but probably should (haven't actually tested changing the owner).

**Comment by hypeJunction** [comment]
Rebased.

**Comment by hypeJunction** [comment]
I wonder if we should also add a listener for `delete` events and wipe the icons before we wipe the entity. Currently if a plugin saves the icon under an owner's directory and forgets to clean icons, we end up with files on the filestore until user account is deleted, and there is no way to scan for them

**Comment by hypeJunction** [comment]
@juho-jaakkola Can you give it a go on your test base?

**Comment by juho-jaakkola** [comment]
I'm swamped with non-Elgg work, but I'll try to find time for testing.

**Comment by hypeJunction** [comment]
Rebased again. @mrclay @juho-jaakkola Let's get this merged and close the icon chapter.

**Comment by mrclay** [comment]
The group icons show up fine for me. What else should we test manually?

**Comment by hypeJunction** [comment]
Changing group owner and making sure all icons relocate to new directory.

File: engine/tests/phpunit/ElggFileTest.php [hunk 2]
```diff
@@ -376,4 +376,43 @@ public function testCanDeleteSymlinkWithMissingTarget() {
...
+	 * @group FileService
+	 */
+	public function testCanTransferFile() {
+
+		$dataroot = elgg_get_config('dataroot');
+
+		$file = new \ElggFile();
+		$file->owner_guid = 3;
+		$file->setFilename("file-to-transfer.txt");
+		$file->setFilename("file-to-transfer.txt");
```
Comment (ewinslow) [inline engine/tests/phpunit/ElggFileTest.php]: Why is this line duplicated?

Comment (mrclay) [inline engine/tests/phpunit/ElggFileTest.php]: Computers, have to tell em twice sometimes.

Comment (hypeJunction) [inline engine/tests/phpunit/ElggFileTest.php]: Humans, have to make them sleep sometimes

**Pull request merged**